### PR TITLE
fixed issues

### PR DESCRIPTION
**Pull request by veners1995**

### CODE REVIEW

**Comment by veners1995** [comment]
@Arthur2e5
Look this, I think it isn't the best way.
So far, I haven't come up with an better idea than this.
So, would you like to come and have a look?   :)

File: ALS/ArchLinuxShell.sh [hunk 1]
```diff
@@ -16,6 +16,40 @@
...
+#桌面环境
+softwareInstallCmd['gnome']='sudo pacman -S --noconfirm gnome'
+softwareInstallCmd['plasma']='sudo pacman -S --noconfirm plasma'
+softwareInstallCmd['xfce4']='sudo pacman -S --noconfirm xfce4'
+softwareInstallCmd['cinnamon']='sudo pacman -S --noconfirm cinnamon\nsudo pacman -S --noconfirm gnome-screenshot\nsudo pacman -S --noconfirm mate-terminal\nsudo pacman -S --noconfirm evince\nsudo pacman -S --noconfirm viewnior'
+softwareInstallCmd['mate']='sudo pacman -S --noconfirm mate'
+
+function chooseSoftware
+{
+	unset choose
```
Comment (Arthur2e5) [inline ALS/ArchLinuxShell.sh]: Call it choice…

**Comment by Arthur2e5** [comment]
1. pacman supports multiple packages after `-S`. Merge them.
2. Using a big dictionary only masks the problem of duplicates. My solution is to define a custom function `pacman_S()` in the resulting shell script so the lines look smaller. `echo bala >> mcv.sh` is also packed.
3. “请输入 1-6” is unnecessary. Use “请输入选择序号” and merge them into your packed choice function. The heading line is also unnecessary, just change it to “选择一款文本编辑器：”
4. The extension packs shall be merged like in my fork.
5. Boolean `read -n 1` operations should be wrapped like `while read -n 1`, or you will miss user keyboard errors.
6. Your usage of case looks not that beautiful. Do some indention.
7. `function xxx` is not suggested. Use `xxx()`.
8. Still, consider gettextization that I have already done. That fixes the problem with tty by providing an English UI, when fbterm is not available. I have mentioned another solution, exec-ing to fbterm which runs `"$0" "$@"` when running in tty in tieba and you can someday implement it.

FInally, @Guanrenfu, if you think that it's nice, just click on the merge button. You can also add @veners1995 to the collaborators of the project in project settings.

File: ALS/ArchLinuxShell.sh [hunk 2]
```diff
@@ -320,135 +336,70 @@ cat << EOF
...
+PS3='请输入1-5:'
+chooseSoftware 'gvim' 'emacs' 'gedit' 'leafpad' '不安装文本编辑器'
+echo "" >> continue.sh
 
-	4)
-		echo "sudo pacman -S --noconfirm leafpad" >> continue.sh
-		echo "" >> continue.sh
-		break
-	;;
+if [ ${choose} == 'gvim' ];then
```
Comment (Arthur2e5) [inline ALS/ArchLinuxShell.sh]: Worse than `case`.

**Comment by Guanrenfu** [comment]
@Arthur2e5 Hum，I had add @veners1995 to collaborators. And however, I couldn't understand your change again...

**Pull request merged**